kivy/uix/filechooser.py

```diff
@@ -811,8 +811,16 @@ def _generate_file_entries(self, *args, **kwargs):
         else:
             if platform == 'win':
                 is_root = splitdrive(path)[1] in (sep, altsep)
+                if not path:
+                    path = abspath(sep)
+                else:
+                    path = abspath(self.path)
             elif platform in ('macosx', 'linux', 'android', 'ios'):
                 is_root = normpath(expanduser(path)) == sep
+                if not path:
+                    path = abspath(sep)
+                else:
+                    path = abspath(self.path)
```

Comment:
Since that code is identical, can't it be moved to after the else and thus deduplicated? Is it really ok to do that on windows?